cmd/testmachinery-run/testrunner/testrunner.go

```diff
@@ -81,96 +71,77 @@ func Run(config *TestrunConfig, parameters *TestrunParameters) {
 	}
 
-	log.Infof("Deploying testrun %s", parameters.TestrunName)
-
-	tmChartRenderer, err = chartrenderer.New(tmClusterClient)
-	if err != nil {
-		log.Fatalf("Cannot create chartrenderer for gardener  %s", err.Error())
-	}
+	finishedTestruns := runChart(tmClient, chart, parameters, metadata.BOM)
 
-	err = common.ApplyChart(tmClusterClient, tmChartRenderer, parameters.TestrunChartPath, parameters.TestrunName, namespace, map[string]interface{}{
-		"testrunName": parameters.TestrunName,
-		"shoot": map[string]interface{}{
-			"name":             parameters.ShootName,
-			"projectNamespace": fmt.Sprintf("garden-%s", parameters.ProjectName),
-			"cloudprovider":    parameters.Cloudprovider,
-			"cloudprofile":     parameters.Cloudprofile,
-			"secretBinding":    parameters.SecretBinding,
-			"region":           parameters.Region,
-			"zone":             parameters.Zone,
-			"k8sVersion":       parameters.K8sVersion,
-			"machinetype":      parameters.MachineType,
-			"autoscalerMin":    parameters.AutoscalerMin,
-			"autoscalerMax":    parameters.AutoscalerMax,
-		},
-		"kubeconfigs": map[string]interface{}{
-			"gardener": string(gardenKubeconfig),
-		},
-	}, map[string]interface{}{})
-	if err != nil {
-		log.Fatalf("Cannot render chart %s", err.Error())
-	}
-
-	log.Infof("Testrun %s deployed", parameters.TestrunName)
-
-	var tr *tmv1beta1.Testrun
-	var testrunPhase argov1.NodePhase
-	startTime := time.Now()
-	for !util.Completed(testrunPhase) {
-		var err error
-
-		if util.MaxTimeExceeded(startTime, maxWaitTimeSeconds) {
-			log.Fatalf("Maximum wait time of %d is exceeded by Testrun %s", maxWaitTimeSeconds, parameters.TestrunName)
+	testrunsFailed := false
+	for _, tr := range finishedTestruns {
+		if config.OutputFile != "" {
+			err = Output(config, tr, metadata)
+			if err != nil {
+				log.Fatal(err.Error())
+			}
 		}
 
-		tr, err = tmClient.Testmachinery().Testruns("default").Get(parameters.TestrunName, metav1.GetOptions{})
-		if err != nil {
-			log.Errorf("Cannot get testrun: %s", err.Error())
+		if config.OutputFile != "" && config.ESConfigName != "" {
+			err = PersistFile(config.OutputFile, config.ESConfigName)
+			if err != nil {
+				log.Errorf("Cannot persist file %s: %s", config.OutputFile, err.Error())
+				return
+			}
 		}
-		testrunPhase = tr.Status.Phase
 
-		if tr.Status.State != "" {
-			log.Infof("Testrun %s is %s. State: %s", parameters.TestrunName, testrunPhase, tr.Status.State)
+		if tr.Status.Phase == argov1.NodeSucceeded {
+			log.Infof("The testrun %s finished successfully", tr.Name)
 		} else {
-			log.Infof("Testrun %s is %s. Waiting ...", parameters.TestrunName, testrunPhase)
+			testrunsFailed = true
+			log.Errorf("Testrun %s failed with phase %s", tr.Name, tr.Status.Phase)
 		}
-
-		time.Sleep(time.Duration(pollIntervalSeconds) * time.Second)
 	}
 
-	err = Output(config, tr, metadata)
-	if err != nil {
-		log.Fatal(err.Error())
+	if config.ESConfigName != "" && config.ConcourseOnErrorDir != "" {
```

Comment:
Why is ESConfigName necessary? Maybe it's better to put this if check into the GenerateNotificationConfigForAlerting. I find this is then more readable

Reply:
done